docs/src/compression.md

```diff
@@ -145,29 +145,47 @@ settings.
 ```julia
 using JLD2
 
-jldopen("example.jld2", "w") do f
+jldopen("example.jld2", "w"; compress=ZstdFilter()) do f
```

Comment:
It seems a bit counter-intuitive that you provide the compress argument to the whole file and then directly overrule it in the first few `write` calls.
Maybe it would be better to move the `default_array` to the top?